nightfall-optimist/src/event-handlers/transaction-submitted.mjs

```diff
@@ -63,8 +63,9 @@ async function transactionSubmittedEventHandler(eventParams) {
     const transactionNullifiers = transaction.nullifiers.filter(
       hash => hash !== '0x0000000000000000000000000000000000000000000000000000000000000000',
     ); // Deposit transactions still have nullifier fields but they are 0
+    const uniqueNullifiers = Array.from(new Set(transactionNullifiers));
```

Comment:
Duplicate nullifiers check has to happen inside `checkTransactionType` of `checkTransaction` call of the `transaction-submitted` event handler. 

When writing the PR description, please specify the various fixes done.